src/test/java/org/gridsuite/study/server/service/dynamicsimulation/DynamicSimulationServiceTest.java

```diff
@@ -217,36 +221,79 @@ public void testGetTimeLineResult() {
 
         // setup timeSeriesClient mock
         // timeline
-        TimeSeriesIndex index = new IrregularTimeSeriesIndex(new long[]{102479, 102479, 102479, 104396});
-        StringTimeSeries timeLine = TimeSeries.createString(TIME_LINE_NAME, index,
-                "CLA_2_5 - CLA : order to change topology",
-                "_BUS____2-BUS____5-1_AC - LINE : opening both sides",
-                "CLA_2_5 - CLA : order to change topology",
-                "CLA_2_4 - CLA : arming by over-current constraint");
-        given(timeSeriesClient.getTimeSeriesGroup(TIME_LINE_UUID, null)).willReturn(Arrays.asList(timeLine));
+        List<TimeLineEventInfos> timeLineEventInfosList = List.of(
+                new TimeLineEventInfos(102479, "CLA_2_5", "CLA : order to change topology"),
+                new TimeLineEventInfos(102479, "_BUS____2-BUS____5-1_AC", "LINE : opening both sides"),
+                new TimeLineEventInfos(102479, "CLA_2_5", "CLA : order to change topology"),
+                new TimeLineEventInfos(104396, "CLA_2_4", "CLA : arming by over-current constraint")
+        );
+        List<TimeSeries> timeLineSeries = new ArrayList<>();
+
+        // collect and convert timeline event list to StringTimeSeries
+        long[] timeLineIndexes = timeLineEventInfosList.stream().mapToLong(event -> (long) event.time()).toArray();
+        String[] timeLineValues = timeLineEventInfosList.stream().map(event -> {
+            try {
+                return objectMapper.writeValueAsString(event);
+            } catch (JsonProcessingException e) {
+                throw new PowsyblException("Error while serializing time line event: " + event.toString(), e);
+            }
+        }).toArray(String[]::new);
+        timeLineSeries.add(TimeSeries.createString("timeLine", new IrregularTimeSeriesIndex(timeLineIndexes), timeLineValues));
+
+        given(timeSeriesClient.getTimeSeriesGroup(TIME_LINE_UUID, null)).willReturn(timeLineSeries);
 
         // call method to be tested
-        List<StringTimeSeries> timeLineResult = dynamicSimulationService.getTimeLineResult(NODE_UUID);
+        List<TimeLineEventInfos> timeLineResult = dynamicSimulationService.getTimeLineResult(NODE_UUID);
 
         // check result
-        // must contain only one
-        assertEquals(1, timeLineResult.size());
+        // must contain 4 timeline events
+        Assertions.assertThat(timeLineResult).hasSize(4);
```

Comment:
May remove `Assertions`

Reply:
DONE! in https://github.com/gridsuite/study-server/pull/532/commits/1466c66f8ef0c06385f90b6c56d218c1c29c9cff